### DISCUSSION TITLE
Auto adjust line-height to adapt row's count

### DISCUSSION THREAD

**Discussion by ofzo** [wezterm/wezterm#5097] [Ideas]
When using the command-line editor, the editor calculates the number of available lines, and the height exceeded will not be used. If the background color configured by the editor is different from the background color of Wezterm, this part will show a strong sense of fragmentation

My idea is whether it is possible to automatically adjust the line-height so that the available height of the wezterm is an integer multiple of the line-height, so that the editor can fully utilize the entire area

![image](https://github.com/wez/wezterm/assets/18000037/502580ee-47c1-423e-868a-468d742f8922)

**Comment by fcying** [comment]
You can dynamically set the line_height according to your needs.
```
function recompute_line_height(window)
    local window_dims = window:get_dimensions()
    local height = window_dims.pixel_height
    local overrides = window:get_config_overrides() or {}
    
    overrides.line_height = 1.0  -- dynamically set
    window:set_config_overrides(overrides)
end

wezterm.on("window-resized", function(window, _)
    recompute_line_height(window)
end)

```
